src/android/aidl/birthday-service/client.md

```diff
@@ -0,0 +1,44 @@
+# AIDL Client
+
+Finally, we can create a Rust client for our new service.
+
+*birthday_service/src/client.rs*:
+
+```rust,ignore
+{{#include birthday_service/src/client.rs:main}}
+```
+
+*birthday_service/Android.bp*:
+
+```javascript
+{{#include birthday_service/Android.bp:birthday_client}}
+```
+
+Notice that the client does not depend on `libbirthdayservice`.
+
+Build, push, and run the client on your device:
+
+```shell
+{{#include ../../build_all.sh:birthday_client}}
+```
+
+```text
+Happy Birthday Charlie, congratulations with the 60 years!
+```
+
+<details>
+
+* `Strong<dyn IBirthdayService>` is the trait object representing the service
+  that the client has connected to.
+    * `Strong` is a custom smart pointer type for Binder. It handles both an
+      in-process ref count for the service trait object, and the global Binder
+      ref count that tracks how many clients are connected to a given service.
```

Comment:
It's more "have a reference to the Binder object," because objects != services.

Reply:
I've tweaked the wording. Let me know if it's still not quite right.